src/Sylius/Component/Core/spec/Promotion/Checker/ContainsProductRuleCheckerSpec.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace spec\Sylius\Component\Core\Promotion\Checker;
+
+use PhpSpec\ObjectBehavior;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\OrderItem;
+use Sylius\Component\Core\Model\ProductVariant;
+use Sylius\Component\Promotion\Exception\UnsupportedTypeException;
+use Sylius\Component\Promotion\Model\PromotionSubjectInterface;
+
+/**
+ * @author Piotr Walków <walkow.piotr@gmail.com>
+ */
+class ContainsProductRuleCheckerSpec extends ObjectBehavior
+{
+    function it_is_initialibable()
+    {
+        $this->shouldHaveType('Sylius\Component\Core\Promotion\Checker\ContainsProductRuleChecker');
+    }
+
+    function it_is_sylius_rule_checker()
+    {
+        $this->shouldImplement('Sylius\Component\Promotion\Checker\RuleCheckerInterface');
+    }
+
+    function it_throws_exception_on_invalid_subject(PromotionSubjectInterface $subject)
+    {
+        $this->shouldThrow(UnsupportedTypeException::class)->duringIsEligible($subject, []);
+    }
+
+    function it_returns_true_if_variant_is_right_and_exclude_is_not_set(
+        OrderInterface $subject,
+        OrderItem $orderItem,
+        ProductVariant $variant
+    ) {
+        $subject->getItems()->shouldBeCalled()->willReturn([$orderItem]);
```

Comment:
`shouldBeCalled()` is useless when you use `willReturn()`. Don't use mocks & stubs to test same thing!